centaur/src/it/scala/centaur/reporting/Slf4jReporter.scala

```diff
@@ -17,22 +17,30 @@ class Slf4jReporter(override val params: ErrorReporterParams)
 
   override lazy val destination: String = "error"
 
-  override def logCentaurFailure(testEnvironment: TestEnvironment,
-                                 ciEnvironment: CiEnvironment,
-                                 centaurTestException: CentaurTestException)
-                                (implicit executionContext: ExecutionContext): IO[Unit] = {
+  override def logFailure(testEnvironment: TestEnvironment,
+                          ciEnvironment: CiEnvironment,
+                          throwable: Throwable)
+                         (implicit executionContext: ExecutionContext): IO[Unit] = {
     IO {
+
+      val errorMessage = throwable match {
+        case centaurTestException: CentaurTestException =>
+          centaurTestException.workflowIdOption.map("with workflow id '" + _ + "' ").getOrElse("")
+        case nonCentaurException =>
+          s"with unexpected non-Centaur exception $nonCentaurException"
+      }
+
       val message =
         s"Test '${testEnvironment.name}' " +
           s"failed on attempt ${testEnvironment.attempt + 1} " +
           s"of ${testEnvironment.retries + 1} " +
-          centaurTestException.workflowIdOption.map("with workflow id '" + _ + "' ").getOrElse("")
+          errorMessage
 
       // Only log fully on the final attempt. Otherwise log a shortened version
       if (testEnvironment.attempt >= testEnvironment.retries) {
-        logger.error(message, centaurTestException)
+        logger.error(message, throwable)
       } else {
-        val messageWithShortExceptionContext = message + " (" + centaurTestException.message.replace("\n", " ").take(150) + "[...])"
+        val messageWithShortExceptionContext = message + " (" + Option(throwable.getMessage).getOrElse("").replace("\n", " ").take(150) + "[...])"
```

Comment:
This no longer crashes when the message is null which is certainly an improvement, though `ExceptionUtils.getMessage(throwable)` (or `getRootCauseMessage`) might be more clear.